src/MySqlConnector/MySqlParameterCollection.cs

```diff
@@ -11,6 +11,13 @@ internal MySqlParameterCollection()
 		m_nameToIndex = new(StringComparer.OrdinalIgnoreCase);
 	}
 
+	public void SetParameterCapacity(int capacity)
+	{
+		if(capacity < 0)
+			throw new ArgumentOutOfRangeException(nameof(capacity));
+		m_parameters.Capacity = capacity;
```

Comment:
```suggestion
		m_parameters.Capacity = capacity;
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP2_1_OR_GREATER
		m_nameToIndex.EnsureCapacity(size);
#endif
```

We should resize the other collection, too.